api/v1beta1/openstackbaremetalset_types.go

```diff
@@ -18,13 +18,27 @@ package v1beta1
 
 import (
 	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // AutomatedCleaningMode is the interface to enable/disable automated cleaning
 // +kubebuilder:validation:Enum=metadata;disabled
 type AutomatedCleaningMode string
 
+// InstanceSpec Instance specific attributes
+type InstanceSpec struct {
+	// +kubebuilder:validation:Optional
+	// CtlPlaneIP - Control Plane IP
+	CtlPlaneIP string `json:"ctlPlaneIP"`
+	// +kubebuilder:validation:Optional
+	// UserData - Host User Data
+	UserData *corev1.SecretReference `json:"userData,omitempty"`
+	// +kubebuilder:validation:Optional
+	// NetworkData - Host Network Data
+	NetworkData *corev1.SecretReference `json:"networkData,omitempty"`
```

Comment:
I'm wondering if it would make sense to append these names with `Secret` so as to make it more obvious that they refer to a `Secret` (like we do we other `Secrets` in other operators).  Same with the similar fields in `OpenStackBaremetalSetSpec`.

Reply:
Would not the type tell the user that it's a SecretReference? I did not see that in metal3-io[1], so kept it the same here. I can change it if we want though.

[1] https://github.com/metal3-io/baremetal-operator/blob/main/apis/metal3.io/v1alpha1/baremetalhost_types.go#L374

Reply:
I suppose so.  But it's not necessarily obvious that it refers to a `Secret` from the YAML, if they happen to not be reading the code (which a user generally wouldn't be doing).  i.e.:

```
                   networkData:
                      description: NetworkData - Host Network Data
                      properties:
                        name:
                          description: name is unique within a namespace to reference
                            a secret resource.
                          type: string
                        namespace:
                          description: namespace defines the space within which the
                            secret name must be unique.
                          type: string
```

But yeah, not important at the moment.  I'm fine with keeping it as-is and just adjusting it later if ambiguity arises.